Src/FluentAssertions/Formatting/PredicateLambdaExpressionValueFormatter.cs

```diff
@@ -0,0 +1,127 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Reflection;
+using System.Runtime.Serialization;
+using System.Text;
+
+namespace FluentAssertions.Formatting
+{
+    /// <summary>
+    /// The <see cref="PredicateLambdaExpressionValueFormatter" /> is responsible for formatting
+    /// boolean lambda expressions.
+    /// </summary>
+    public class PredicateLambdaExpressionValueFormatter : IValueFormatter
+    {
+        public bool CanHandle(object value) => value is LambdaExpression lambdaExpression && lambdaExpression.ReturnType == typeof(bool);
+
+        public string Format(object value, FormattingContext context, FormatChild formatChild)
+        {
+            var lambdaExpression = value as LambdaExpression;
+
+            var reducedExpression = ReduceConstantSubExpressions(lambdaExpression.Body);
+
+            if (reducedExpression is BinaryExpression binaryExpression && binaryExpression.NodeType == ExpressionType.AndAlso)
+            {
+                var subExpressions = ExtractChainOfExpressionsJoinedWithAndOperator(binaryExpression);
+                return string.Join(" AndAlso ", subExpressions.Select(_ => _.ToString()));
+            }
+
+            return reducedExpression.ToString();
+        }
+
+        /// <summary>
+        /// This step simplifies the lambda expression by replacing parts of it which do not depend on the lambda parameters
+        /// with the actual values of these sub-expressions. The simplified expression is much easier to read. 
+        /// E.g. "(_.Text == "two") AndAlso (_.Number == 3)"
+        /// Instead of "(_.Text == value(FluentAssertions.Specs.Collections.GenericCollectionAssertionsSpecs+c__DisplayClass122_0).twoText) AndAlso (_.Number == 3)".
+        /// </summary>
+        private static Expression ReduceConstantSubExpressions(Expression expression)
+        {
+            return new ConstantSubExpressionReductionVisitor().Visit(expression);
+        }
+
+        /// <summary>
+        /// This step simplifies the lambda expression by removing unnecessary parentheses for root level chain of AND operators.
+        /// E.g. (_.Text == "two") AndAlso (_.Number == 3) AndAlso (_.OtherText == "foo")
+        /// Instead of ((_.Text == "two") AndAlso (_.Number == 3)) AndAlso (_.OtherText == "foo")
+        /// This simplification is only implemented for the chain of AND operators beacuse this is the most common predicate scenario.
```

Comment:
```suggestion
        /// This simplification is only implemented for the chain of AND operators because this is the most common predicate scenario.
```